src/hooks/useUser/useUser.tsx

```diff
@@ -0,0 +1,109 @@
+import { useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+
+import { getExpoToken } from '@/context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import { useFeatureFlag } from '@/hooks/useFeatureFlag/useFeatureFlag'
+import { isMobile } from '@/lib/globals'
+import { getOrCreateUser, registerDevice } from '@/lib/user/api'
+import type { User } from '@/lib/user/types'
```

Comment:
ditto, wondering if we should use a better name across the board since that's super generic?